trin-execution/src/block_reward.rs

```diff
@@ -24,18 +30,51 @@ pub fn get_block_reward(block: &ProcessedBlock) -> Vec<(Address, u128)> {
     if base_block_reward > 0 {
         if let Some(uncles) = &block.uncles {
             for uncle in uncles.iter() {
-                beneficiaries.push((
-                    uncle.author,
-                    ((8 + uncle.number - block.header.number) as u128 * base_block_reward) >> 3,
-                ));
+                *beneficiaries.entry(uncle.author).or_default() +=
+                    ((8 + uncle.number - block.header.number) as u128 * base_block_reward) >> 3;
             }
 
-            beneficiaries.push((
-                block.header.author,
-                base_block_reward + (base_block_reward >> 5) * uncles.len() as u128,
-            ));
+            *beneficiaries.entry(block.header.author).or_default() +=
+                base_block_reward + (base_block_reward >> 5) * uncles.len() as u128;
+        }
+    }
+}
+
+pub fn process_withdrawals(block: &ProcessedBlock, beneficiaries: &mut HashMap<Address, u128>) {
+    if let Some(withdrawals) = &block.withdraws {
+        for withdrawal in withdrawals.iter() {
+            if withdrawal.amount == 0 {
+                continue;
+            }
+            *beneficiaries.entry(withdrawal.address).or_default() += withdrawal.amount as u128;
         }
     }
+}
+
+fn process_dao_fork(
```

Comment:
nit: I just want to highlight that for me, this doesn't fit here. It's not a "reward", the way others are.
But considering recent discussion, I think you have strong feeling about it. So I don't think it's worth discussing this. Feel free to ignore this comment.